### PR TITLE
Pantry funct tj

### PR DESCRIPTION
**Pull request by taylorjohnson141**
## Description

This PR gives the user the ability to check each recipe and see if the user has the right amount of ingredients to cook the recipe as well as telling the user what ingredients they are missing. 

Clearly and concisely describe the feature.

User can click on a recipe image and has a pop up to see which ingredients the user is missing as well as the amount of ingredients the user is missing for their pantry. 

## Affected areas of application

domUpdates.js Scripts.js
List all parts of the website that will be altered from this PR.
What is displayed on the pop up

## How Has This Been Tested?
npm run-scripts build
npm start
Running the webpack server and clicking a recipe image and seeing the popup.

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Test A

- Test B

## Relevant Tickets

What tickets/issues were resolved by this feature

- Closes <...>

## Screenshots
Screen Shot 2020-09-16 at 1 51 53 PM

Add screenshots here if applicable

### CODE REVIEW

File: src/css/styles.scss [hunk 1]
```diff
@@ -45,6 +45,7 @@ body {
     width: 80%;
     border-radius: 0.5rem;
     overflow: scroll;
+    padding: 3rem;
```
Comment (CalebCyphers) [inline src/css/styles.scss:48]: Thanks for adding this padding in, I think it helps readability a lot

File: src/domUpdates.js [hunk 2]
```diff
@@ -35,20 +35,38 @@ let domUpdates = {
...
 
   },
 
-  displayRecipeInfo(recipe, ingredients) {
+  displayRecipeInfo(recipe,cookables,missinIng) {
+    let cookable; 
+    
+    let isRecipeCookable = cookables.find(recipe =>{
+      return recipe.name === recipe
+    }) !== undefined
```
Comment (CalebCyphers) [inline src/domUpdates.js:47]: Nice use of 'find' and bang operators here

File: src/domUpdates.js [hunk 3]
```diff
@@ -63,22 +81,19 @@ let domUpdates = {
...
   },
   
-   displayPantry(pantry) {
-    if(pantryArea === null) {
-      return 
-    }
+  displayPantry(pantry) {
     pantry = pantry.contents
-    pantryArea.innerHTML = ''
+    recipeCards.innerHTML = ''
```
Comment (CalebCyphers) [inline src/domUpdates.js:94]: What is the reason for this change? Is it just for readability?

File: src/domUpdates.js [hunk 4]
```diff
@@ -144,137 +159,23 @@ let domUpdates = {
...
-//     ${ingredient.quantity.amount.toFixed(2)} ${ingredient.quantity.unit}
-//     ${ingredient.name}</li></ul>
-//     `)
-//   })
-//   recipeObject.instructions.forEach(instruction => {
-//     instructionsSpan.insertAdjacentHTML('beforebegin', `<li>
-//     ${instruction.instruction}</li>
-//     `)
-//   })
-// }
```
Comment (CalebCyphers) [inline src/domUpdates.js:243]: Thanks for cleaning up this commented-out stuff. Next time make sure to mention this in your description of the PR though.

**Review by CalebCyphers — APPROVED** [review]

File: src/domUpdates.js [hunk 5]
```diff
@@ -35,20 +35,38 @@ let domUpdates = {
...
+      return recipe.name === recipe
+    }) !== undefined 
+
+ 
+    if (isRecipeCookable) {
+      
+      cookable = 'You can cook this recipe!'
+    } else {
+      cookable = `Can't cook this recipe! <br>
+      You are missing: <br>${this.formatObjectsToDisplatCorrectly((this.figureMissing(recipe, missinIng)))}`
```
Comment (nicolegooden) [inline src/domUpdates.js:55]: I think a good refactor opportunity would be to have these invocations restructured.  Not sure how this could get done...but, it's a little hard to keep track of which values are getting passed in on line 55.

File: src/domUpdates.js [hunk 6]
```diff
@@ -35,20 +35,38 @@ let domUpdates = {
...
-
+    }).join('<br>')
+    recipeDisplay.innerHTML = ''
     recipeDisplay.innerHTML += 
     `<section class="recipe-display-main">
+    <h1>${cookable}</h1>
     <h1>${recipe.name}</h1>
     <div class="recipe-ingredients-with-cost">
-      <p class="popup-ingredients">${JSON.stringify(recipe.ingredients)}</p>
+    <h2>Nessasary Ingredient</h2>
```
Comment (nicolegooden) [inline src/domUpdates.js:68]: maybe this should be `Ingredients` on line 68 to account for multiple ingredients being shown (more readable)

File: src/domUpdates.js [hunk 7]
```diff
@@ -144,137 +159,23 @@ let domUpdates = {
...
-//     `)
-//   })
-// }
+    })
+  },
+  figureMissing(recipe, missinIng) {
+    let correctRecipe = missinIng.find(rep =>{
+      return recipe.name === rep.name
+    }) 
+    console.log(correctRecipe)
```
Comment (nicolegooden) [inline src/domUpdates.js:168]: we should clean up this `console.log` as well as the others in any other files!

File: src/domUpdates.js [hunk 8]
```diff
@@ -144,137 +159,23 @@ let domUpdates = {
...
-// function getFavorites() {
-//   if (user.favoriteRecipes.length) {
-//     user.favoriteRecipes.forEach(recipe => {
-//       document.querySelector(`.favorite${recipe.id}`).classList.add('favorite-active')
-//     })
-//   } else return
-// }
+    return correctRecipe.NotEnough
+  },
+  formatObjectsToDisplatCorrectly(obj) {
```
Comment (nicolegooden) [inline src/domUpdates.js:172]: Should this be `formatObjectsToDisplayCorrectly()` ? I think there is a typo.
Also on 175, it might be more easily read by the user if the ingredient name is shown on the left, with the amount and unit shown to the right of the ingredient's name.

**Review by nicolegooden — APPROVED** [review]
